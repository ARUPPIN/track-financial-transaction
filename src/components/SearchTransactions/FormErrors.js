import React from 'react'
import { camelCaseToTitle } from '../../common/utils'
export const FormErrors = ({ formErrors }) => (
  <div className='formErrors'>
    {Object.keys(formErrors).map((fieldName, i) => {
      if (formErrors[fieldName].length > 0) {
        return (
          <p key={i} align='center' style={{ color: 'red' }}>
            {camelCaseToTitle(fieldName)} {formErrors[fieldName]}
          </p>
        )
      } else {
        return ''
      }
    })}
  </div>
)
