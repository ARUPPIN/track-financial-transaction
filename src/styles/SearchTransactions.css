.container-search {
  max-width: 500px;
  margin: 50px auto;
}
.search-field {
  margin: 0 0 1em 0;
}
.search-field-label {
  width: 30%;
}
.search-field-input {
  width: 70%;
}
.field-block {
  display: block;
}
.field-inline-block {
  display: inline-block;
}
.form-btn-align {
  text-align: center;
}
