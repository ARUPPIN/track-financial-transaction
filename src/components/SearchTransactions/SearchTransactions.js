"use strict";
import React from "react";
import { Button, Form, FormGroup } from "reactstrap";
import Datetime from "react-datetime";
import { FormErrors } from "./FormErrors";
import { convertMillisToDate, isValidDate } from "../../common/utils";
import { ACCOUNT_ID_FORMAT } from "../../common/constants";
import "../../styles/SearchTransactions.css";

class SearchTransactions extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      accountId: "",
      fromDate: "",
      toDate: "",
      formErrors: { accountId: "", fromDate: "", toDate: "" },
      accountIdValid: false,
      fromDateValid: false,
      toDateValid: false,
      formValid: false
    };
    this.searchTransaction = this.searchTransaction.bind(this);
    this.handleAccountIdChg = this.handleAccountIdChg.bind(this);
    this.handleFromDateChg = this.handleFromDateChg.bind(this);
    this.handleToDateChg = this.handleToDateChg.bind(this);
    this.validateChange = this.validateChange.bind(this);
  }

  validateChange(name, val) {
    let { formErrors, accountIdValid, fromDateValid, toDateValid } = this.state;
    switch (name) {
      case "accountId":
        var patt = new RegExp(ACCOUNT_ID_FORMAT);
        accountIdValid = patt.test(val);
        formErrors.accountId = accountIdValid ? "" : " is invalid";
        break;
      case "fromDate":
        fromDateValid = isValidDate(val);
        formErrors.fromDate = fromDateValid ? "" : " is invalid";
        break;
      case "toDate":
        toDateValid = isValidDate(val);
        formErrors.toDate = toDateValid ? "" : " is invalid";
        break;
      default:
        break;
    }
    this.setState(
      {
        formErrors: formErrors,
        accountIdValid: accountIdValid,
        fromDateValid: fromDateValid,
        toDateValid: toDateValid
      },
      this.validateForm
    );
  }

  validateForm() {
    this.setState({
      formValid:
        this.state.accountIdValid &&
        this.state.fromDateValid &&
        this.state.toDateValid
    });
  }

  handleAccountIdChg(e) {
    if (e && e.target) {
      e.preventDefault();
      const name = e.target.name;
      const value = e.target.value;
      //Add sufficient validations
      this.setState({ accountId: e.target.value.toString() }, () => {
        this.validateChange(name, value);
      });
    }
  }

  handleFromDateChg(e) {
    const date = e;
    this.setState({ fromDate: convertMillisToDate(e) }, () => {
      this.validateChange("fromDate", date);
    });
  }

  handleToDateChg(e) {
    const date = e;
    this.setState({ toDate: convertMillisToDate(e) }, () => {
      this.validateChange("toDate", date);
    });
  }

  searchTransaction(e) {
    e.preventDefault();
    this.props.searchTransactionsInfo(
      this.state.accountId,
      this.state.fromDate,
      this.state.toDate
    );
  }

  render() {
    const { accountId, fromDate, toDate, formValid } = this.state;
    return (
      <>
        <Form className="container-search" onSubmit={this.searchTransaction}>
          <FormGroup>
            <label className="search-field-label">Account Id: </label>
            <input
              name="accountId"
              className="search-field-input"
              type="text"
              value={accountId}
              onChange={this.handleAccountIdChg}
              required
              placeholder="Enter Account Id"
            />
          </FormGroup>
          <FormGroup>
            <label>From Date: </label>
            <i className="far fa-calendar-alt"></i>
            <Datetime
              dateFormat="DD/MM/YYYY"
              timeFormat="HH:mm:ss"
              timeConstraints={{
                hours: { min: 0, max: 23 },
                minutes: { min: 0, max: 59 },
                seconds: { min: 0, max: 59 }
              }}
              closeOnTab={true}
              closeOnSelect={true}
              value={fromDate}
              onChange={this.handleFromDateChg}
              inputProps={{
                placeholder: "Select transaction start date",
                required: true,
                className: "field-from-date"
              }}
            />
          </FormGroup>
          <FormGroup>
            <label>To Date: </label>
            <Datetime
              dateFormat="DD/MM/YYYY"
              timeFormat="HH:mm:ss"
              timeConstraints={{
                hours: { min: 0, max: 23 },
                minutes: { min: 0, max: 59 },
                seconds: { min: 0, max: 59 }
              }}
              closeOnTab={true}
              closeOnSelect={true}
              value={toDate}
              onChange={this.handleToDateChg}
              inputProps={{
                placeholder: "Select transaction end date",
                required: true,
                className: "field-to-date"
              }}
            />
          </FormGroup>
          <FormGroup className="form-btn-align">
            {formValid ? (
              <Button outline color="primary" type="submit">
                Search Transactions
              </Button>
            ) : (
              <Button outline disabled>
                Search Transactions
              </Button>
            )}
          </FormGroup>
          <FormGroup>
            <div className="panel panel-default">
              <FormErrors formErrors={this.state.formErrors} />
            </div>
          </FormGroup>
        </Form>
      </>
    );
  }
}

export default SearchTransactions;
