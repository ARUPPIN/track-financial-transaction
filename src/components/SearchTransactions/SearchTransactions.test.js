'use strict'
import React from 'react'
import { configure, mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import ReactSixteenAdapter from 'enzyme-adapter-react-16'
import SearchTransactions from './SearchTransactions'
configure({ adapter: new ReactSixteenAdapter() })

describe('SearchTransactions', () => {
  const searchTransactionsInfo = jest.fn()
  const component = mount(
    <SearchTransactions searchTransactionsInfo={searchTransactionsInfo} />
  )
  it('snapshot renders', () => {
    let tree = mountToJson(component, { mode: 'deep' })
    expect(tree).toMatchSnapshot()
  })

  it('component did mount with form and submit button', () => {
    expect(component.find('form').length).toEqual(1)
    const submitBtn = component.find('button')
    expect(submitBtn.length).toEqual(1)
    expect(submitBtn.props().disabled).toBe(true)
  })

  it("component's accountId is updated", () => {
    const accountIdInput = component.find('.search-field-input')
    expect(accountIdInput.length).toEqual(1)
    accountIdInput.simulate('change', {
      target: { value: 'ACC334455', name: 'accountId' },
      preventDefault: jest.fn()
    })
    expect(component.state('accountId')).toBe('ACC334455')
    expect(component.state('accountIdValid')).toBe(true)
  })

  it("component's accountId is not updated", () => {
    const accountIdInput = component.find('.search-field-input')
    expect(accountIdInput.length).toEqual(1)
    accountIdInput.simulate('change', {
      target: { value: '123456789', name: 'accountId' },
      preventDefault: jest.fn()
    })
    expect(component.state('accountIdValid')).toBe(false)
  })

  it("component's fromDate is updated", () => {
    const fromDateInput = component.find('.field-from-date')
    expect(fromDateInput.length).toEqual(1)
    component.instance().handleFromDateChg('2016-10-19')
    expect(component.state('fromDateValid')).toBe(true)
  })

  it("component's fromDate is not updated", () => {
    const fromDateInput = component.find('.field-from-date')
    expect(fromDateInput.length).toEqual(1)
    component.instance().handleFromDateChg('')
    expect(component.state('fromDateValid')).toBe(false)
  })

  it("component's toDate is updated", () => {
    const toDateInput = component.find('.field-to-date')
    expect(toDateInput.length).toEqual(1)
    component.instance().handleToDateChg('2016-10-19')
    expect(component.state('toDateValid')).toBe(true)
  })

  it("component's toDate is not updated", () => {
    const toDateInput = component.find('.field-to-date')
    expect(toDateInput.length).toEqual(1)
    component.instance().handleToDateChg('')
    expect(component.state('toDateValid')).toBe(false)
    expect(component.state('formValid')).toBe(false)
  })

  it('component formValid test', () => {
    expect(component.state('formValid')).toBe(false)
  })

  it('component searchTransaction test', () => {
    let evt = {
      target: { value: '123456789' },
      preventDefault: jest.fn()
    }
    component.instance().searchTransaction(evt)
    expect(searchTransactionsInfo).toHaveBeenCalled()
  })

  it('component submit button is disabled', () => {
    expect(component.find('form').length).toEqual(1)
    const submitBtn = component.find('button')
    expect(submitBtn.length).toEqual(1)
    expect(submitBtn.props().disabled).toBe(true)
  })
})
