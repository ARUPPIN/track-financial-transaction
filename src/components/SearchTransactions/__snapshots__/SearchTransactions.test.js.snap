// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTransactions snapshot renders 1`] = `
<form
  className="container-search"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="search-field-label"
    >
      Account Id: 
    </label>
    <input
      className="search-field-input"
      name="accountId"
      onChange={[Function]}
      placeholder="Enter Account Id"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      From Date: 
    </label>
    <div
      className="rdt"
    >
      <input
        className="form-control field-from-date"
        key="i"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select transaction start date"
        required={true}
        type="text"
        value=""
      />
      <div
        className="rdtPicker"
        key="dt"
      >
        <div
          className="rdtDays"
        >
          <table>
            <thead
              key="th"
            >
              <tr
                key="h"
              >
                <th
                  className="rdtPrev"
                  key="p"
                  onClick={[Function]}
                >
                  <span>
                    ‹
                  </span>
                </th>
                <th
                  className="rdtSwitch"
                  colSpan={5}
                  data-value={9}
                  key="s"
                  onClick={[Function]}
                >
                  October 2019
                </th>
                <th
                  className="rdtNext"
                  key="n"
                  onClick={[Function]}
                >
                  <span>
                    ›
                  </span>
                </th>
              </tr>
              <tr
                key="d"
              >
                <th
                  className="dow"
                  key="Su0"
                >
                  Su
                </th>
                <th
                  className="dow"
                  key="Mo1"
                >
                  Mo
                </th>
                <th
                  className="dow"
                  key="Tu2"
                >
                  Tu
                </th>
                <th
                  className="dow"
                  key="We3"
                >
                  We
                </th>
                <th
                  className="dow"
                  key="Th4"
                >
                  Th
                </th>
                <th
                  className="dow"
                  key="Fr5"
                >
                  Fr
                </th>
                <th
                  className="dow"
                  key="Sa6"
                >
                  Sa
                </th>
              </tr>
            </thead>
            <tbody
              key="tb"
            >
              <tr
                key="10_5"
              >
                <td
                  className="rdtDay rdtOld"
                  data-value={29}
                  key="9_29"
                  onClick={[Function]}
                >
                  29
                </td>
                <td
                  className="rdtDay rdtOld"
                  data-value={30}
                  key="9_30"
                  onClick={[Function]}
                >
                  30
                </td>
                <td
                  className="rdtDay"
                  data-value={1}
                  key="10_1"
                  onClick={[Function]}
                >
                  1
                </td>
                <td
                  className="rdtDay"
                  data-value={2}
                  key="10_2"
                  onClick={[Function]}
                >
                  2
                </td>
                <td
                  className="rdtDay rdtToday"
                  data-value={3}
                  key="10_3"
                  onClick={[Function]}
                >
                  3
                </td>
                <td
                  className="rdtDay"
                  data-value={4}
                  key="10_4"
                  onClick={[Function]}
                >
                  4
                </td>
                <td
                  className="rdtDay"
                  data-value={5}
                  key="10_5"
                  onClick={[Function]}
                >
                  5
                </td>
              </tr>
              <tr
                key="10_12"
              >
                <td
                  className="rdtDay"
                  data-value={6}
                  key="10_6"
                  onClick={[Function]}
                >
                  6
                </td>
                <td
                  className="rdtDay"
                  data-value={7}
                  key="10_7"
                  onClick={[Function]}
                >
                  7
                </td>
                <td
                  className="rdtDay"
                  data-value={8}
                  key="10_8"
                  onClick={[Function]}
                >
                  8
                </td>
                <td
                  className="rdtDay"
                  data-value={9}
                  key="10_9"
                  onClick={[Function]}
                >
                  9
                </td>
                <td
                  className="rdtDay"
                  data-value={10}
                  key="10_10"
                  onClick={[Function]}
                >
                  10
                </td>
                <td
                  className="rdtDay"
                  data-value={11}
                  key="10_11"
                  onClick={[Function]}
                >
                  11
                </td>
                <td
                  className="rdtDay"
                  data-value={12}
                  key="10_12"
                  onClick={[Function]}
                >
                  12
                </td>
              </tr>
              <tr
                key="10_19"
              >
                <td
                  className="rdtDay"
                  data-value={13}
                  key="10_13"
                  onClick={[Function]}
                >
                  13
                </td>
                <td
                  className="rdtDay"
                  data-value={14}
                  key="10_14"
                  onClick={[Function]}
                >
                  14
                </td>
                <td
                  className="rdtDay"
                  data-value={15}
                  key="10_15"
                  onClick={[Function]}
                >
                  15
                </td>
                <td
                  className="rdtDay"
                  data-value={16}
                  key="10_16"
                  onClick={[Function]}
                >
                  16
                </td>
                <td
                  className="rdtDay"
                  data-value={17}
                  key="10_17"
                  onClick={[Function]}
                >
                  17
                </td>
                <td
                  className="rdtDay"
                  data-value={18}
                  key="10_18"
                  onClick={[Function]}
                >
                  18
                </td>
                <td
                  className="rdtDay"
                  data-value={19}
                  key="10_19"
                  onClick={[Function]}
                >
                  19
                </td>
              </tr>
              <tr
                key="10_26"
              >
                <td
                  className="rdtDay"
                  data-value={20}
                  key="10_20"
                  onClick={[Function]}
                >
                  20
                </td>
                <td
                  className="rdtDay"
                  data-value={21}
                  key="10_21"
                  onClick={[Function]}
                >
                  21
                </td>
                <td
                  className="rdtDay"
                  data-value={22}
                  key="10_22"
                  onClick={[Function]}
                >
                  22
                </td>
                <td
                  className="rdtDay"
                  data-value={23}
                  key="10_23"
                  onClick={[Function]}
                >
                  23
                </td>
                <td
                  className="rdtDay"
                  data-value={24}
                  key="10_24"
                  onClick={[Function]}
                >
                  24
                </td>
                <td
                  className="rdtDay"
                  data-value={25}
                  key="10_25"
                  onClick={[Function]}
                >
                  25
                </td>
                <td
                  className="rdtDay"
                  data-value={26}
                  key="10_26"
                  onClick={[Function]}
                >
                  26
                </td>
              </tr>
              <tr
                key="11_2"
              >
                <td
                  className="rdtDay"
                  data-value={27}
                  key="10_27"
                  onClick={[Function]}
                >
                  27
                </td>
                <td
                  className="rdtDay"
                  data-value={28}
                  key="10_28"
                  onClick={[Function]}
                >
                  28
                </td>
                <td
                  className="rdtDay"
                  data-value={29}
                  key="10_29"
                  onClick={[Function]}
                >
                  29
                </td>
                <td
                  className="rdtDay"
                  data-value={30}
                  key="10_30"
                  onClick={[Function]}
                >
                  30
                </td>
                <td
                  className="rdtDay"
                  data-value={31}
                  key="10_31"
                  onClick={[Function]}
                >
                  31
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={1}
                  key="11_1"
                  onClick={[Function]}
                >
                  1
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={2}
                  key="11_2"
                  onClick={[Function]}
                >
                  2
                </td>
              </tr>
              <tr
                key="11_9"
              >
                <td
                  className="rdtDay rdtNew"
                  data-value={3}
                  key="11_3"
                  onClick={[Function]}
                >
                  3
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={4}
                  key="11_4"
                  onClick={[Function]}
                >
                  4
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={5}
                  key="11_5"
                  onClick={[Function]}
                >
                  5
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={6}
                  key="11_6"
                  onClick={[Function]}
                >
                  6
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={7}
                  key="11_7"
                  onClick={[Function]}
                >
                  7
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={8}
                  key="11_8"
                  onClick={[Function]}
                >
                  8
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={9}
                  key="11_9"
                  onClick={[Function]}
                >
                  9
                </td>
              </tr>
            </tbody>
            <tfoot
              key="tf"
            >
              <tr>
                <td
                  className="rdtTimeToggle"
                  colSpan={7}
                  onClick={[Function]}
                >
                  00:00:00
                </td>
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label>
      To Date: 
    </label>
    <div
      className="rdt"
    >
      <input
        className="form-control field-to-date"
        key="i"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select transaction end date"
        required={true}
        type="text"
        value=""
      />
      <div
        className="rdtPicker"
        key="dt"
      >
        <div
          className="rdtDays"
        >
          <table>
            <thead
              key="th"
            >
              <tr
                key="h"
              >
                <th
                  className="rdtPrev"
                  key="p"
                  onClick={[Function]}
                >
                  <span>
                    ‹
                  </span>
                </th>
                <th
                  className="rdtSwitch"
                  colSpan={5}
                  data-value={9}
                  key="s"
                  onClick={[Function]}
                >
                  October 2019
                </th>
                <th
                  className="rdtNext"
                  key="n"
                  onClick={[Function]}
                >
                  <span>
                    ›
                  </span>
                </th>
              </tr>
              <tr
                key="d"
              >
                <th
                  className="dow"
                  key="Su0"
                >
                  Su
                </th>
                <th
                  className="dow"
                  key="Mo1"
                >
                  Mo
                </th>
                <th
                  className="dow"
                  key="Tu2"
                >
                  Tu
                </th>
                <th
                  className="dow"
                  key="We3"
                >
                  We
                </th>
                <th
                  className="dow"
                  key="Th4"
                >
                  Th
                </th>
                <th
                  className="dow"
                  key="Fr5"
                >
                  Fr
                </th>
                <th
                  className="dow"
                  key="Sa6"
                >
                  Sa
                </th>
              </tr>
            </thead>
            <tbody
              key="tb"
            >
              <tr
                key="10_5"
              >
                <td
                  className="rdtDay rdtOld"
                  data-value={29}
                  key="9_29"
                  onClick={[Function]}
                >
                  29
                </td>
                <td
                  className="rdtDay rdtOld"
                  data-value={30}
                  key="9_30"
                  onClick={[Function]}
                >
                  30
                </td>
                <td
                  className="rdtDay"
                  data-value={1}
                  key="10_1"
                  onClick={[Function]}
                >
                  1
                </td>
                <td
                  className="rdtDay"
                  data-value={2}
                  key="10_2"
                  onClick={[Function]}
                >
                  2
                </td>
                <td
                  className="rdtDay rdtToday"
                  data-value={3}
                  key="10_3"
                  onClick={[Function]}
                >
                  3
                </td>
                <td
                  className="rdtDay"
                  data-value={4}
                  key="10_4"
                  onClick={[Function]}
                >
                  4
                </td>
                <td
                  className="rdtDay"
                  data-value={5}
                  key="10_5"
                  onClick={[Function]}
                >
                  5
                </td>
              </tr>
              <tr
                key="10_12"
              >
                <td
                  className="rdtDay"
                  data-value={6}
                  key="10_6"
                  onClick={[Function]}
                >
                  6
                </td>
                <td
                  className="rdtDay"
                  data-value={7}
                  key="10_7"
                  onClick={[Function]}
                >
                  7
                </td>
                <td
                  className="rdtDay"
                  data-value={8}
                  key="10_8"
                  onClick={[Function]}
                >
                  8
                </td>
                <td
                  className="rdtDay"
                  data-value={9}
                  key="10_9"
                  onClick={[Function]}
                >
                  9
                </td>
                <td
                  className="rdtDay"
                  data-value={10}
                  key="10_10"
                  onClick={[Function]}
                >
                  10
                </td>
                <td
                  className="rdtDay"
                  data-value={11}
                  key="10_11"
                  onClick={[Function]}
                >
                  11
                </td>
                <td
                  className="rdtDay"
                  data-value={12}
                  key="10_12"
                  onClick={[Function]}
                >
                  12
                </td>
              </tr>
              <tr
                key="10_19"
              >
                <td
                  className="rdtDay"
                  data-value={13}
                  key="10_13"
                  onClick={[Function]}
                >
                  13
                </td>
                <td
                  className="rdtDay"
                  data-value={14}
                  key="10_14"
                  onClick={[Function]}
                >
                  14
                </td>
                <td
                  className="rdtDay"
                  data-value={15}
                  key="10_15"
                  onClick={[Function]}
                >
                  15
                </td>
                <td
                  className="rdtDay"
                  data-value={16}
                  key="10_16"
                  onClick={[Function]}
                >
                  16
                </td>
                <td
                  className="rdtDay"
                  data-value={17}
                  key="10_17"
                  onClick={[Function]}
                >
                  17
                </td>
                <td
                  className="rdtDay"
                  data-value={18}
                  key="10_18"
                  onClick={[Function]}
                >
                  18
                </td>
                <td
                  className="rdtDay"
                  data-value={19}
                  key="10_19"
                  onClick={[Function]}
                >
                  19
                </td>
              </tr>
              <tr
                key="10_26"
              >
                <td
                  className="rdtDay"
                  data-value={20}
                  key="10_20"
                  onClick={[Function]}
                >
                  20
                </td>
                <td
                  className="rdtDay"
                  data-value={21}
                  key="10_21"
                  onClick={[Function]}
                >
                  21
                </td>
                <td
                  className="rdtDay"
                  data-value={22}
                  key="10_22"
                  onClick={[Function]}
                >
                  22
                </td>
                <td
                  className="rdtDay"
                  data-value={23}
                  key="10_23"
                  onClick={[Function]}
                >
                  23
                </td>
                <td
                  className="rdtDay"
                  data-value={24}
                  key="10_24"
                  onClick={[Function]}
                >
                  24
                </td>
                <td
                  className="rdtDay"
                  data-value={25}
                  key="10_25"
                  onClick={[Function]}
                >
                  25
                </td>
                <td
                  className="rdtDay"
                  data-value={26}
                  key="10_26"
                  onClick={[Function]}
                >
                  26
                </td>
              </tr>
              <tr
                key="11_2"
              >
                <td
                  className="rdtDay"
                  data-value={27}
                  key="10_27"
                  onClick={[Function]}
                >
                  27
                </td>
                <td
                  className="rdtDay"
                  data-value={28}
                  key="10_28"
                  onClick={[Function]}
                >
                  28
                </td>
                <td
                  className="rdtDay"
                  data-value={29}
                  key="10_29"
                  onClick={[Function]}
                >
                  29
                </td>
                <td
                  className="rdtDay"
                  data-value={30}
                  key="10_30"
                  onClick={[Function]}
                >
                  30
                </td>
                <td
                  className="rdtDay"
                  data-value={31}
                  key="10_31"
                  onClick={[Function]}
                >
                  31
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={1}
                  key="11_1"
                  onClick={[Function]}
                >
                  1
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={2}
                  key="11_2"
                  onClick={[Function]}
                >
                  2
                </td>
              </tr>
              <tr
                key="11_9"
              >
                <td
                  className="rdtDay rdtNew"
                  data-value={3}
                  key="11_3"
                  onClick={[Function]}
                >
                  3
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={4}
                  key="11_4"
                  onClick={[Function]}
                >
                  4
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={5}
                  key="11_5"
                  onClick={[Function]}
                >
                  5
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={6}
                  key="11_6"
                  onClick={[Function]}
                >
                  6
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={7}
                  key="11_7"
                  onClick={[Function]}
                >
                  7
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={8}
                  key="11_8"
                  onClick={[Function]}
                >
                  8
                </td>
                <td
                  className="rdtDay rdtNew"
                  data-value={9}
                  key="11_9"
                  onClick={[Function]}
                >
                  9
                </td>
              </tr>
            </tbody>
            <tfoot
              key="tf"
            >
              <tr>
                <td
                  className="rdtTimeToggle"
                  colSpan={7}
                  onClick={[Function]}
                >
                  00:00:00
                </td>
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-btn-align form-group"
  >
    <button
      aria-label={null}
      className="btn btn-outline-secondary disabled"
      disabled={true}
      onClick={[Function]}
    >
      Search Transactions
    </button>
  </div>
  <div
    className="form-group"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="formErrors"
      />
    </div>
  </div>
</form>
`;
